core/include/detray/tools/navigator.hpp

```diff
@@ -260,27 +254,71 @@ class navigator {
     DETRAY_HOST_DEVICE
     navigator(const detector_t &d) : _detector(&d) {}
 
+    template <typename track_t>
+    DETRAY_HOST_DEVICE inline intersection update_next_candidate(
```

Comment:
I like this function, but can we please try to roll back on the ```target()``` and ```status()``` function? The ```stepper``` should set the correct trust level in the navigator state instead, if possible. If there is a bug, we should try to fix it, but I must admit I am not a fan of changing the structure of the class

Reply:
Roll backing is OK but why don't you like it..?

 >  The stepper should set the correct trust level in the navigator state instead, if possible

I don't understand this line.

Reply:
I find it much more readable concerning what the idea of the navigator was (at least that's what I understood when I first read it): It is a cache of candidates that is updated depending on the trust level which is set in the navigator state. The trust level reflects how much trust we have that the data in the cache is still useful, for example after an external change of the track state. It makes therefore sense to me to have a single update function that calls the different update strategies according to the trust level. The ```target()``` and ```status()``` calls are an interface that make it ACTS compatible, but do not reflect the underlying data structure, imo.
Depending on the change they made, the actors should request how the candidates cache is updated: Have I changed the state so much that we should re-evaluate everything? Or is it enough to just re-evaluate the (previously) closest surface? For example, a small step might only need to update the distance to the currently targeted surface. For a very large step we might want to re-evaluate everything, but not sort again or whatever

Maybe it turns out that this idea is not actually useful?

Reply:
Isn't it also possible to do the same in the new changes that I made?
If the trust level drops fair trust level by the actor or whatever, we can call `set_next` in target call by adding a condition for `fair_trust`. But it doesn't have to be called in status call.

Reply:
I am just not convinced that the same update function is called in two different function. Why don't we remove target call if it is just status()?

Reply:
And thing is that everything works same as before in this PR. What I did was just simple decomposition of `update_kernel`. I tried to make`target()` target surface and `status()` update status. In the context of that, I am not satisfied yet but it makes much more sense than before to me :/..

Reply:
Maybe I need to think about it more by reading some ACTS codes to fully understand the idea

Reply:
It is right, i also don't see the difference between ```target()``` and ```status()``` in detray, because detrays navigator implementation is different. That's what I asked Andi today. We only have these functions, because that's the interface with ACTS, as far as I know. I don't see any other reason to keep them, either. It just makes no sense to me, to adapt the navigator to this structure by splitting up the update function when it does not fit what the detray navigator as a data structure does. But if you insist, we can go ahead, because, sure it will probably work